mantle/kola/tests/misc/network.go

```diff
@@ -273,6 +283,17 @@ var (
           primary_ip=%s
           secondary_ip=%s
 
+          connections_folder=$(NetworkManager --print-config | sed -nr "/^\[keyfile\]/ { :l /^path[ ]*=/ { s/.*=[ ]*//; p; q;}; n; b l;}")
+          if [[ -z $connections_folder ]]; then
```

Comment:
@RamLavi  not sure about this, your checking if a folder exists shouldn't it be -d  ? 
And if its true, then you don't want to change it right? so you do need to set the path 

I think this should be: 
if [[ ! -d $connections_folder ]]; then 
       connections_folder=/etc/NetworkManager/system-connections
fi

Reply:
You're right, but I think we should first check if the string is not empty.

Reply:
So should it be with "not (!)" ?

Reply:
ahh I missread you so let me rephrase.
`connections_folder` is a string, we need to:
1. check if the string is not empty (using the `-z` condition: `[[ -z $connections_folder ]]`)
2. (following your suggestion) check that `connections_folder` directory actually exists. (using the `-d` condition: `[[ ! -d $connections_folder ]]`
if the conditions fail then we must use the default path value.

The reason that condition 2 is not enough is that `-d` can return true if the var is not set:
```
# [ -d $SOME_UNSET_VAR ] && echo true || echo false
true
```
but according to [this](https://superuser.com/questions/655926/why-does-checking-whether-a-folder-exists-using-d-on-an-empty-string-return-tru) we can also do `[[ ! -d "$connections_folder" ]]` and we'll be fine

Reply:
changed to `[[ ! -d "$connections_folder" ]]`

Reply:
Cool, I think you can close this review